### PR TITLE
lock dependencies in package.json

### PR DESCRIPTION
**Pull request by zan-kusterle**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -10209,42 +10244,42 @@
...
+        "uuid": "^3.1.0"
       }
     },
     "request-promise-core": {
       "version": "1.1.1",
       "resolved": "http://npm.celtra.com/request-promise-core/-/request-promise-core-1.1.1.tgz",
       "integrity": "sha1-Pu4AssWqgyOc+wTFcA2jb4HNCLY=",
       "requires": {
-        "lodash": "4.17.10"
+        "lodash": "^4.13.1"
```
Comment (kurent) [inline package-lock.json]: you can see that it did the same... Thats some weird behaviour from package-lock.json. I think we should just create npm-shrinkwrap.json as we know thats what we actually want. 

just run ```npm shrinkwrap```

https://docs.npmjs.com/cli/shrinkwrap
https://docs.npmjs.com/files/package-lock.json

**Comment by kurent** [comment]
hm, @zan-kusterle are we sure we want to use yarn? I get warning when doing yarn install:
You are using Node "10.0.0" which is not supported and may encounter bugs or unexpected behavior. Yarn supports the following semver range: "^4.8.0 || ^5.7.0 || ^6.2.2 || ^8.0.0"

**Comment by kurent** [comment]
[This thread](https://github.com/npm/npm/issues/20434) explains this npm@6 change a bit. Someone suggested setting up devDependency of npm@6. 

> So what we did for our project is added npm as a dev dependency to the package.json. By that after the first install, the user will use npm v6 and we committed the changes to the package-lock.json. With that, we found a suitable workaround for us. But this might not work in all environments.

**Comment by zan-kusterle** [comment]
Interesting. I also discovered a killer feature of Yarn which is that it actually respects the `engines` field in `package.json`. But I guess we can keep npm and just add it as a dev dependency...

**Comment by zan-kusterle** [comment]
Also, `yarn install` works fine on my machine with node 10.5.0.

**Comment by kurent** [comment]
I see, updating yarn solves this issue. Yea, I guess in this case, yarn is the best option to go with. We have both requirements:
1. consistent lock file for dependencies
2. enforcing of yarn version

@3buson

**Pull request merged**